ivre/parser/iptables.py

```diff
@@ -0,0 +1,81 @@
+#! /usr/bin/env python
+
+# This file is part of IVRE.
+# Copyright 2011 - 2018 Pierre LALET <pierre.lalet@cea.fr>
+#
+# IVRE is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# IVRE is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+# or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with IVRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# Maintainers:
+#   - 2018 Francois CHENAIS <francois.chenais@cea.fr>
+#
+
+"""Support for Iptables log from syslog files
+
+
+Feb  4 05:30:11 pi01 kernel: [3240403.495065] IPTABLES/UDP/IN=enxb827eb8f8a4f OUT= MAC=ff:ff:ff:ff:ff:ff:18:62:2c:7e:45:d0:08:00:45:00:00:ec:00:00:40:00:40:11:df:e3 SRC=192.168.0.254 DST=192.168.0.255 LEN=236 TOS=0x00 PREC=0x00 TTL=64 ID=0 DF PROTO=UDP SPT=138 DPT=138 LEN=216
+
+TODO:
+ - the log does not contain the Year so it's necessary to set it using parameter
+   or changing input with new field 'YEAR' or using current year.
+ - The log contains mac addr from source and destination, is this inifo can be
+   added to neo db ?
+
+"""
+
+import datetime
+from builtins import zip
+from ivre.parser import Parser
+
+
+class Iptables(Parser):
+    """Iptables log generator from a syslog file descriptor"""
+
+    def __init__(self, fname, pcap_filter=None):
+        super(Iptables, self).__init__(fname)
+
+
+    @classmethod
+    def parse_line(self, line):
+        line = line.decode().rstrip('\r\n')
+        if not line:
+            return next(self)
+
+        fields={}
```

Comment:
I prefer using comprehension expressions when possible. Don't you think this would work?
```python
fields = dict(val.split('=', 1) if '=' in val else (val, "") for val in line[line.find('IN='):].split())
```

Also, maybe it would be better to check if 'IN=' appears in the line (and return `next(self)` if it does not). What do you think?

And, given the following lines, maybe we could use the opportunity to `.lower()` the keys:

```python
fields = dict((key.lower(), value) for key, value in (val.split('=', 1) if '=' in val else (val, "") for val in line[line.find('IN='):].split()))
```

Reply:
For the return condition, I use the example of argus and suppose that the input file contains only netfilter logs according to the FAQ example.  Your suggestion would allow parsing files without previous filter processing. It's fine !

For the choice of comprehension expression, U are the master, I do like the master want !:D 

I do the change.